condition.go

```diff
@@ -258,75 +287,118 @@ const (
+		return "", errors.New("to construct a query, Name must be specified")
 	}
 
-	if (condition.Field == "" && condition.Label == "") && condition.FieldManipulationFunction == "" {
-		return "", errors.New("field, function or label can not be empty")
+	if condition.Label == "" && (condition.Field == "" && condition.FieldManipulationFunction == "") {
+		return "", errors.New("either Label or (Field, FieldManipulationFunction) must be specified")
 	}
 
-	if condition.Field != "" && condition.Label != "" && condition.FieldManipulationFunction != "" {
-		return "", errors.New("field and label can not both be defined")
+	if condition.Field != "" && condition.Label != "" {
+		return "", errors.New("only one of Field and Label can be specified")
 	}
 
-	query := ""
+	if condition.FieldManipulationFunction != "" && condition.Field == "" {
+		return "", errors.New("if FieldManipulationFunction is specified, Field must also be specified")
+	}
+
+	if condition.Label != "" {
+		sb.WriteString(fmt.Sprintf("%s:%s", condition.Name, condition.Label))
+		return sb.String(), nil
+	}
 
-	//build the fields
+	var node string
 	if condition.Field != "" {
-		query += fmt.Sprintf("%s.%s", condition.Name, condition.Field)
-	} else if condition.Label != "" {
-		//we're done here
-		return fmt.Sprintf("%s:%s", condition.Name, condition.Label), nil
+		node = fmt.Sprintf("%s.%s", condition.Name, condition.Field)
 	} else {
-		query = condition.Name
+		node = condition.Name
 	}
 
 	if condition.FieldManipulationFunction != "" {
-		query = fmt.Sprintf("%s(%s)", condition.FieldManipulationFunction, query)
+		sb.WriteString(fmt.Sprintf("%s(%s)", condition.FieldManipulationFunction, node))
+	} else {
+		sb.WriteString(node)
 	}
 
 	if condition.ConditionOperator == "" && condition.ConditionFunction == "" {
-		return "", errors.New("either condition operator or condition function has to be defined")
+		return "", errors.New("one of ConditionOperator or ConditionFunction must be specified")
 	}
 
 	if condition.ConditionOperator != "" && condition.ConditionFunction != "" {
-		return "", errors.New("operator and function can not both be defined")
+		return "", errors.New("only one of ConditionOperator or ConditionFunction can be specified")
 	}
 
-	//build the operators
-	if condition.ConditionOperator != "" {
-		query += fmt.Sprintf(" %s", condition.ConditionOperator)
-	} else if condition.ConditionFunction != "" {
-		//if its a condition function, we're done
-		return fmt.Sprintf("%s(%s)", condition.ConditionFunction, query), nil
+	// Handle ConditionFunction if specified
+	if condition.ConditionFunction != "" {
+		if condition.NegateCondition {
+			return fmt.Sprintf(
+				"NOT %s(%s)", condition.ConditionFunction, strings.Trim(sb.String(), "NOT "),
+			), nil
+		}
+		return fmt.Sprintf("%s(%s)", condition.ConditionFunction, sb.String()), nil
 	}
 
-	//check if its valid for in
+	// Add ConditionOperator to query
+	sb.WriteString(fmt.Sprintf(" %s", condition.ConditionOperator))
```

Comment:
Nit: `strings.Builder` implements `io.Writer` so you could simply use `fmt.Fprintf`:
```suggestion
	fmt.Fprintf(sb, " %s", condition.ConditionOperator)
```

Alternatively, it would be even more efficient to just write the elements directly:
```suggestion
	sb.WriteRune(' ')
	sb.WriteString(condition.ConditionOperator.String())
```

Same in the other places where you call `sb.WriteString(fmt.Sprintf(...))`.

Reply:
Updated to use WriteRune + WriteString